src/sys/mod.rs

```diff
@@ -5,17 +5,19 @@
 //! `event`: a module with various helper functions for `Event`, see
 //!          `crate::event::Event` for the required functions.
 
-#[cfg(unix)]
-pub use self::unix::{
-    event, Event, Events, Selector, SocketAddr, SourceFd, TcpListener, TcpStream, UdpSocket,
-    UnixDatagram, UnixListener, UnixStream, Waker,
-};
+#[cfg(all(unix, feature = "os-poll"))]
+pub use self::unix::*;
 
-#[cfg(unix)]
+#[cfg(all(unix, feature = "os-poll"))]
 mod unix;
 
-#[cfg(windows)]
-pub use self::windows::{event, Event, Events, Selector, TcpListener, TcpStream, UdpSocket, Waker};
+#[cfg(all(windows, feature = "os-poll"))]
+pub use self::windows::*;
 
-#[cfg(windows)]
+#[cfg(all(windows, feature = "os-poll"))]
 mod windows;
+
+#[cfg(not(feature = "os-poll"))]
+mod shell;
+#[cfg(not(feature = "os-poll"))]
+pub(crate) use self::shell::*;
```

Comment:
Just a question; can we use the `compile_error` macro here? I don't if that work when a crate depends on Mio without features though.

Reply:
Well. That is a good question. The goal is that libs **dont** pull in the os-poll feature but end executables do. If it is a compile error, running cargo check in the lib would fail. Is there a way to handle that?

Reply:
After a quick test: it fails to compile/check with `cargo check` on a crate the doesn't use `os-poll`, using `compile_error`.

Reply:
Maybe we can use something like https://github.com/japaric/panic-never, that fails a link time. But that can be done later.